Update README.md (#39)
Lomadriel/WaitFreeHashMap: README.md

WIP

## Requirements

- Compiler which supports C++17
- CMake ≥ 3.10

> ## Minimal exemple

Reviewer: ```suggestion
## Minimal unordered map example
```

```cpp
#include <iostream>
#include <thread>

#include <wfc/unordered_map.hpp>

constexpr int nbr_threads = 16;

int main()
{
    wfc::unordered_map<std::size_t, std::size_t> m(4, nbr_threads, nbr_threads);

    std::array<std::thread, nbr_threads> threads;
    for (std::size_t i = 0; i < nbr_threads; ++i)
    {
        threads[i] = std::thread([&m, i]() {
            m.insert(i, i);
        });
    }

    for (auto& t: threads)
    {
        t.join();
    }

    m.visit([](std::pair<const std::size_t, std::size_t> p) {
        std::cout << '[' << p.first << '-' << p.second << "]\n";
    });

    return 0;
}
```

This should output:

```
[0-0]
[1-1]
[2-2]
[3-3]
[4-4]
[5-5]
[6-6]
[7-7]
[8-8]
[9-9]
[10-10]
[11-11]
[12-12]
[13-13]
[14-14]
[15-15]
```

## How to import the library using CMake

> To include the library, you may copy / paste everything in a subfolder (such as `externals`) or use a git submodule.

Reviewer: ```suggestion
To include the library, you may copy / paste the content of this repository in a subfolder (such as `externals`) or use a git submodule.
```

The build targets in this repository are for tests, exemples and code formatting.
These are good to know should you contribute to this project or play with the exemples.

First, you need to create a sub-directory to run `cmake` from it. Then you can build using `make`.

```
$ mkdir build
$ cmake .. -DWFC_BUILD_ALL=1
```

By default, everything is build in Release mode. You can pass `-DCMAKE_BUILD_TYPE=Debug` to `cmake` to ask Debug builds.
This will also activate useful features for developing purposes such as the holy warnings.
The `-DWFC_BUILD_ALL=1` parameter will tell CMake to include all our targets in the build process.

CMake will generate several targets that you can build separately using the `--target` parameter:

- `UnorderedMapExample1`
- `UtilityTests`
- `UnorderedMapTests`
- `Clang-format`

For instance, to build the `UnorderedMapTests` run 

```
$ cmake --build . --target UnorderedMapTests
```

You can also build everything by not giving any specific target.
Produced executables are inside the `bin` folder.

Note that the `Clang-format` target does not produce anything.
It just run the clang formatter.

> Also the target needs to be called explicitly.

Reviewer: ```suggestion
Also this target needs to be called explicitly.
```
